docs/developers/connectors/requirements/task3.md

```diff
@@ -637,6 +613,24 @@ Below are the property functions specific to the `Derivative` class.
 | `funding_payment_span` | `List[int]`           | Returns the `_funding_payment_span` instance variable representing the time span (in seconds) before and after funding period when exchanges consider active positions eligible for funding payment. `_funding_payment_span` can be set on initialization. |
 | `position_mode`        | `PositionMode`        | Returns the current position mode for exchanges that support both one-way and hedge modes.                                                                                                                                                                 |
 
+### ClientOrderTracker and the InFlightOrder Class
+
+As seen in the [Connector Component Overview](/developers/connectors/architecture/#connector-component-overview), the `Exchange`/`Derivative` class depends on the `ClientOrderTracker` and the `InFlightOrder` Class.
+The `InFlightOrder` abstracts an order's details and is primarily used by the `Exchange`/`Derivative` class through the `ClientOrderTracker` to manage all active orders.
+
+The **_InFlightOrder and ClientOrderTracker Class Diagram_**, given below, details the critical variables, properties and methods in the `InFlightOrder` and `ClientTracker` class and the relationship between them.
+
+![InFlightOrderClientOrderTrackerUMLDiagram](/assets/img/in-flight-order-client-order-tracker-class-diagram.svg)
+
+!!! note
+The `InFlightOrder` in a `ClientOrderTracker` associated with a `Derivative` class takes advantage of the `leverage` and `position` attributes.
+
+If not in a `Derivative` class, the default `leverage` is set to 1 and default `position` is set to NIL.
+
+The `position` attribute is of type [`PositionAction`](https://github.com/hummingbot/hummingbot/blob/master/hummingbot/core/event/events.py#L51-L54).
+
+The `InFlightOrder` class and the `ClientOrderTracker` class are intended to be used as they are, universally, without any need to be extended or modified through inheritance.
+
```

Comment:
I think it will be useful to include when to use the `process_order_update()` and `process_trade_update()` in the `NewConnector` class.

i.e. `process_order_update()` should really be called to update the order status, while `process_trade_update()` should be used to include an executed trade on a particular order.

Reply:
I've added the two methods.